### ISSUE TITLE
<Fill/> width can exceed <ViewPort/> width

### ISSUE THREAD

**Issue by Slapbox** [aeagle/react-spaces#107]
**Describe the bug**
In versions `0.2.6` and `0.3.2`, using `react-spaces` as a controlled component, we're seeing an issue where resizing an anchored space on the left side of the screen to a smaller size can result in a situation like this:

**window.innerWidth:** 940px
**Viewport width:** 940px
**Left Anchored width:** 96px - although React state shows `anchorSize: 34`
**Fill space width:** 1035px

Note that 940px + 96px = 1036px ~== 1035px, perhaps a hint as to the cause

**To Reproduce**
I'm not entirely sure how to reproduce this. Is there a CodeSandbox to use as a jumping off point?

**Expected behavior**
All spaces should remain within the rendered portion of the viewport and the combined widths should not exceed `window.innerWidth`

**Screenshots**

![image](https://user-images.githubusercontent.com/6835891/143113724-4e398f12-e8f4-41dd-95d7-f055bcb989e7.png)

![image](https://user-images.githubusercontent.com/6835891/143114739-4a41813c-c47b-4ca4-a5df-405a5e65399a.png)

**Package versions (please complete the following information):**
 - React version: 17.0.2
 - React spaces version: 0.2.6 & 0.3.2

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 7/10
 - Browser [e.g. chrome, safari] Electron
 - Version [e.g. 22] 11.4.x

 
**Code sandbox link**
None available currently

**Additional context**
Thank you for your work on this increasingly awesome project!

**Comment by aeagle** [comment]
PR #139

**Comment by aeagle** [comment]
@Slapbox. I seemed to have tracked down the cause of this issue to a race condition in the resize operation. Looking at the code there was a possibility that the handler which deals with mouse/touch moves during a resize operation could also fire after the resize operation had completed and also after the `onResizeEnd` callback. This would result in the out of sync UI that you have seen.

I've put a fix in `v0.4.1` which now hopefully fixes this issue.

**Comment by Slapbox** [comment]
Thanks for your awesome work @aeagle!

I can't recall the last time I saw this issue so maybe whatever I was experiencing - I think we found a workaround for our use-case because we're on `0.3.6` currently and I haven't seen the issue in ages. As such, unfortunately I can't confirm the fix on my end - but I think this issue can be considered resolved!

Apologies for forgetting to update the issue, and a big thanks to you for not forgetting about it!
